climpyrical/tests/test_gridding.py

```diff
@@ -34,72 +36,72 @@ def test_check_ndims(data, n, passed, error):
-            ValueError,
-        ),
-        (
-            np.linspace(-33.8800048828125, 33.8800048828125, 155),
-            np.linspace(0, 10, 130),
-            False,
-            ValueError,
-        ),
-        (xi, yi, True, None),
-        (xext, yext, True, None),
+        ("x", yi, ds, False, TypeError),
+        (xi, "y", ds, False, TypeError),
+        (np.linspace(0, 10, 30), yi, ds, False, ValueError),
+        (xi, np.linspace(0, 10, 130), ds, False, ValueError),
+        (xi, yi, ds, True, None),
+        (xext_ex, yext_ex, ds, True, None),
     ],
 )
-def test_check_input_coords(x, y, passed, error):
+def test_check_input_coords(x, y, ds, passed, error):
     if passed:
-        check_input_coords(x, y)
+        check_input_coords(x, y, ds)
     else:
         with pytest.raises(error):
-            check_input_coords(x, y)
+            check_input_coords(x, y, ds)
 
 
 @pytest.mark.parametrize(
-    "x,y,xext,yext,passed",
+    "x,y,xext_ex,yext_ex,ds,passed,error",
     [
-        (xi, yi, xext, np.delete(yext, xi.size), False),
-        (xi, np.delete(yi, 2), xext, yext, False),
-        (xi, yi, xext, yext, True),
-        (xi, yi, xext_bad, yext, False),
-        (xi, yi, xext, yext_bad, False),
-        (yi, xi, xext, yext, False),
-        (yi, xi, yext, xext, False),
+        (xi, yi, xext_ex, np.delete(yext_ex, xi.size), ds, False, ValueError),
+        (xi, np.delete(yi, 2), xext_ex, yext_ex, ds, False, ValueError),
+        (xi, yi, xext_ex, yext_ex, ds, True, None),
+        (xi, yi, xext_bad, yext_ex, ds, False, ValueError),
+        (xi, yi, xext_ex, yext_bad, ds, False, ValueError),
+        (yi, xi, xext_ex, yext_ex, ds, False, ValueError),
+        (yi, xi, yext_ex, xext_ex, ds, False, ValueError),
     ],
 )
-def test_check_coords_are_flattened(x, y, xext, yext, passed):
+def test_check_coords_are_flattened(x, y, xext_ex, yext_ex, ds, passed, error):
     if passed:
-        check_coords_are_flattened(x, y, xext, yext)
+        check_coords_are_flattened(x, y, xext_ex, yext_ex, ds)
     else:
-        with pytest.raises((TypeError, ValueError)):
-            check_coords_are_flattened(x, y, xext, yext)
+        with pytest.raises(error):
+            check_coords_are_flattened(x, y, xext_ex, yext_ex, ds)
 
 
-@pytest.mark.parametrize("x,y,xext,yext", [(xi, yi, xext, yext)])
-def test_flatten_coords(x, y, xext, yext):
-    xx, yx = flatten_coords(x, y)
-    assert np.array_equal(xext, xx) and np.array_equal(yext, yx)
+@pytest.mark.parametrize(
+    "xi,yi,xext_ex,yext_ex,ds",
+    [(xi, yi, xext_ex, yext_ex, ds)]
+)
+def test_flatten_coords(xi, yi, xext_ex, yext_ex, ds):
+    xext, yext = flatten_coords(xi, yi, ds)
+    assert np.array_equal(xext_ex, xext) and np.array_equal(yext_ex, yext)
 
 
 x_station, y_station = np.linspace(-100, -80, 10), np.linspace(45, 70, 10)
 x_station_bad, y_station_bad = (
     np.linspace(-10, 10, 10),
-    np.linspace(-10, 10, 10),
+    np.linspace(-10, 10, 10)
```

Comment:
Totally subjective thing here, but I've never been a fan of assignments of the form `x, y = a, b`, except in a couple of special cases. And you've actually used up 2 extra lines in this particular case compared to the simpler and more direct (my opinion)

```
x_station_bad = np.linspace(-10, 10, 10)
y_station_bad = np.linspace(-10, 10, 10)
```